File: BBB.Framework.Dto/Core/EncounterDto.cs
namespace BBB.Framework.Dto.Core
{
    public sealed class EncounterDto
    {
        public Guid EncounterKey { get; set; }

        public string Name { get; set; } = string.Empty;

        public string? Description { get; set; }

        public string? Preparation { get; set; }

        public string? Fight { get; set; }

        public string? Danger { get; set; }

        public int SortOrder { get; set; }

        public Guid RaidKey { get; set; }

        public List<EncounterPositionDto> Positions { get; set; } = new();
    }
}

File: BBBRaiderWeb/src/models/EncounterDto.ts
import {
  logTypeMissmatch,
  parseArray,
  parseNumber,
  parseString,
} from '../helpers/model'
import { EncounterPositionDto } from './EncounterPositionDto'

export class EncounterDto {
  encounterKey: string = ''
  name: string = ''
  description: string = ''
  preparation: string = ''
  fight: string = ''
  danger: string = ''
  sortOrder: number = 0
  raidKey: string = ''
  positions: EncounterPositionDto[] = []

  static fromJSON(obj: any): EncounterDto {
    if (!obj) {
      logTypeMissmatch('EncounterDto', obj)
      return new EncounterDto()
    }

    return {
      encounterKey: parseString(obj.encounterKey),
      name: parseString(obj.name),
      description: parseString(obj.description),
      preparation: parseString(obj.preparation),
      fight: parseString(obj.fight),
      danger: parseString(obj.danger),
      sortOrder: parseNumber(obj.sortOrder),
      raidKey: parseString(obj.raidKey),
      positions: parseArray(obj.positions, EncounterPositionDto.fromJSON),
    }
  }
}
